app/models/unique_deck_type.rb

```diff
@@ -30,6 +30,50 @@ def self.get_type_popularity(time_ago)
     arche.update(arche){|type, wins| wins.to_f/tot * 100 }
   end
 
+  def self.get_top_decks
+    decks = {}
+    arch_pop = Rails.cache.read('archetype_pop').sort_by{|name, val| val}.reverse
+
+    ar1 = []
+    ar2 = []
+    ar3 = []
+
+    @ar1 = UniqueDeckType.where(name: arch_pop[0][0])[0].unique_decks.where("num_matches >= ?", "30").sort_by{|ud| ud.winrate ? ud.winrate : 0 }.reverse
+    @ar1.each do |ud|
+      x = ud.decks.where(is_public: true).where("user_num_matches >= ?", "30").sort_by{|ud| ud.user_winrate ? ud.user_winrate : 0 }
+      unless x.last(1)[0].nil?
+        ar1 << x.last(1)[0]
+      end
+    end
+    ar1 = ar1.first(8)
+    ar1_name = arch_pop[0][0]
+    decks[ar1_name] = ar1
+
+    @ar2 = UniqueDeckType.where(name: arch_pop[1][0])[0].unique_decks.where("num_matches >= ?", "30").sort_by{|ud| ud.winrate ? ud.winrate : 0 }.reverse
+    @ar2.each do |ud|
+      x = ud.decks.where(is_public: true).where("user_num_matches >= ?", "30").sort_by{|ud| ud.user_winrate ? ud.user_winrate : 0 }
+      unless x.last(1)[0].nil?
+        ar2 << x.last(1)[0]
+      end
+    end
+    ar2 = ar2.first(8)
+    ar2_name = arch_pop[1][0]
+    decks[ar2_name] = ar2
+
+    @ar3 = UniqueDeckType.where(name: arch_pop[2][0])[0].unique_decks.where("num_matches >= ?", "30").sort_by{|ud| ud.winrate ? ud.winrate : 0 }.reverse
+    @ar3.each do |ud|
+      x = ud.decks.where(is_public: true).where("user_num_matches >= ?", "30").sort_by{|ud| ud.user_winrate ? ud.user_winrate : 0 }
+      unless x.last(1)[0].nil?
+        ar3 << x.last(1)[0]
+      end
+    end
+    ar3 = ar3.first(8)
+    ar3_name = arch_pop[2][0]
+    decks[ar3_name] = ar3
+
```

Comment:
DRY!

The archtype code is EXACTLY the same for all three archetypes. Extract this to a private method and use it here for better code.

It'll then be:
`
Arches = Get UniqueDecks (Through one query if possible)
Arches.each do 
  ExtractInfo(arche)
`